torch/_decomp/decompositions.py

```diff
@@ -1376,3 +1376,88 @@ def upsample_bilinear2d_vec(
     q2 = torch.mul(v3, xscale1) + torch.mul(v4, xscale2)
     result = torch.mul(q1, yscale1) + torch.mul(q2, yscale2)
     return result
+
+
+# We should be applying decompositions after all transformations
+@register_decomposition(aten.is_same_size.default)
+def is_same_size(a: Tensor, b: Tensor) -> bool:
+    return a.shape == b.shape
+
+
+@register_decomposition(aten.nll_loss_forward)
+def nll_loss_forward(
+    self: Tensor,
+    target: Tensor,
+    weight: Optional[Tensor],
+    reduction: int,
+    ignore_index: int,
+) -> Tuple[Tensor, Tensor]:
+    assert self.dim() > 0 and self.dim() <= 2, "input tensor should be 1D or 2D"
+    assert (
+        target.dim() <= 1
+    ), "0D or 1D target tensor expected, multi-target not supported"
+
+    no_batch_dim = self.dim() == 1 and target.dim() == 0
+    assert no_batch_dim or (
+        self.shape[0] == target.shape[0]
+    ), f"size mismatch (got input: {self.shape}, target: {target.shape})"
+
+    n_classes = self.shape[-1]
+
+    assert weight is None or (
+        weight.dim() == 1 and weight.numel() == n_classes
+    ), f"weight tensor should be defined either for all {n_classes} classes or no classes but got weight tensor of shape: {weight.shape}"  # noqa: B950
+
+    # self can be [N, C] or [C]
+    # target can be [N] or []
+
+    n_dims = self.dim()
+    channel_dim = 1
+    if n_dims < 2:
+        channel_dim = 0
+
+    if weight is not None:
+        # Here is a specific case with reduction mean and non-batched tensors
+        # https://github.com/pytorch/pytorch/issues/61309
+        # In this case weight is cancelled: w * x[t] / w -> x[t]
+        if not (reduction == Reduction.MEAN.value and n_dims < 2):
+            w = weight.unsqueeze(0) if n_dims > 1 else weight
```

Comment:
would this special case be automatically handled if you didn't later have `if result.dim() > 0` ?

Reply:
Seems like it passes the tests if I change this. @vfdev-5 (who originally wrote this decomp)?

Reply:
I guess maybe the idea is that since it cancels out you don't need this term?

Reply:
Yeah but that's microoptimization at the expense of more complicated code? no batch size should already be pretty rare.

Reply:
I don't see any issue with removing that (cc: @vfdev-5 if you see anything wrong with the modified version).